File: python/sorting/SelectionSort.py
#selction sort
def selectionSort(array):
    for i in range(len(array)-1):
        for j in range(i+1, len(array)):
            if array[i] > array[j]:
                array[j], array[i] = array[i], array[j]

if __name__ == "__main__":
    array = [4, 5, 12, 5, 78, 16, 34, 1, 90, -90]
    selectionSort(array)
    print(array)

File: c/sorting/selectionSort.c
/* Selection sort */

#include<stdio.h>
void selectionSort(int arr[], int size) {
        for(int i=0; i<size; i++) {
            int min = i;
            for(int j=i; j<size; j++) {
                if(arr[j] < arr[min]) 
                    min = j;
            }
            int temp = arr[i];
            arr[i] = arr[min];
            arr[min] = temp;
        }
}
int main(void) {
    int arr[] = {5, 3, 9, 1, 2, 8, 4, 1};
    int size = sizeof(arr) / sizeof(arr[0]);
    selectionSort(arr, size);
    for(int i=0; i<size; i++) {
        printf("%d\t", arr[i]);
    }
    return 0;
}
/*
-----Output-----
1       1       2       3       4       5       8       9
*/